### ISSUE TITLE
Using current ssh environment when no username set

### ISSUE THREAD

**Issue by ra-albrecht** [tenstad/terraform-provider-remote#181]
Hi,,

I tried to use terraform for updating certificates on a remote host. Fort his ssh is required. Problem I have: 

The remote provider enforces setting a user name. This can be a problem when multiple users work in the same environment because they have different usernames on a machine.

I am trying to understand why this is necessary. ssh knows much better how to handle this. E.g., when no username is set the connection string should not be 

@<remotehost>:22

but 

remotehost

(the need for the :22 when it is the standard port I don't understand, too, but ok)

The current behavior requires one extra configuration step, which wouldn't be needed when the standard behavior of ssh would be used. 

example:

```hcl
provider "remote" {
  alias = "swarm-master"
  max_sessions = 2
  conn {
    host = "swarm.XXXX.com"
    user = "remoteuser"
  }
}
```

could be (and should I think) be possible as 

```hcl
provider "remote" {
  alias = "swarm-master"
  max_sessions = 2
  conn {
    host = "swarm.XXXX.com"
  }
}
```

This would keep things in large environments much easier. Most times the users have in their personal ssh-config the correct username already set for a remote. 

Would this change possible?

**Comment by gthieleb** [comment]
You can workaround this with setting a variable in your shell session that exposes the current username. 

```
export TF_VAR_remote_user=$USER
```

```
provider "remote" {
  alias = "swarm-master"
  max_sessions = 2
  conn {
    host = "swarm.XXXX.com"
    user = var.remote_user
  }
}

**Comment by ra-albrecht** [comment]
Yes, of course, this is possible. But as said, it's an extra configuration step that should not be required. A fully configured SSH environment mostly exists and should be used instead.
In larger environments, this could make things much easier. Or when you use multiple accounts on multiple hosts you already set in your ssh-config.
